<!doctype html>
<html lang="en">

  <head>
    <meta charset="utf-8">

    <title>A Bioinformatics Paper Review - by Craig Sketchley</title>

    <meta name="author" content="Craig Sketchley">

    <meta name="apple-mobile-web-app-capable" content="yes" />
    <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent" />

    <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">

    <link rel="stylesheet" href="css/reveal.min.css">
    <link rel="stylesheet" href="css/theme/moon.css" id="theme">

    <!-- For syntax highlighting -->
    <link rel="stylesheet" href="lib/css/zenburn.css">

    <!-- If the query includes 'print-pdf', include the PDF print sheet -->
    <script>
      if( window.location.search.match( /print-pdf/gi ) ) {
        var link = document.createElement( 'link' );
        link.rel = 'stylesheet';
        link.type = 'text/css';
        link.href = 'css/print/pdf.css';
        document.getElementsByTagName( 'head' )[0].appendChild( link );
      }
    </script>

    <!--[if lt IE 9]>
    <script src="lib/js/html5shiv.js"></script>
    <![endif]-->

    <!-- Custom text alignment -->
    <style type="text/css">
    h1, h2, h3, h4, h5, h6, p, li{ text-align: left; }


    </style>

  </head>



  <body>

    <div class="reveal">
      <div class="slides">

        <section id="titlePage">
          <h2>A Bioinformatics Paper Review</h2>
            <img class="titleImg" src="images/article.png">
          <p>
            <small>By Craig Sketchley</small>
          </p>

          <aside class="notes" data-markdown>
            # The Paper

            I am going to be discussing the paper shown here:
            - STAR: ultrafast universal RNA-seq aligner

            Note: Do I need an overview slide of the talk?
          </aside>
        </section>

        <section id="introduction">

          <section id="RNAseq">
            <h2>Motivation</h2>
            <h3 class="fragment roll-in">The Problem: RNA-Seq</h3>
            <img class="fragment roll-in" width="300px" align="right" src="images/rna-seq2.png">
            <p class="fragment roll-in">RNA-seq: sequencing the mRNA within a cell at a given point in time:
              <ul>
                <li class="fragment roll-in">Identify where in the genome</li>
                <li class="fragment roll-in">Characterisation of gene expression</li>
              </ul>
            </p>
            <p class="fragment roll-in">High-throughput sequencing challenging to detect &amp; characterise spliced transcripts</p>
            <p class="fragment roll-in">Two key tasks:
              <ul>
                <li class="fragment roll-in">Accurate alignment</li>
                <li class="fragment roll-in">Mapping sequences from non-contiguous regions</li>
              </ul>
            </p>
            <p class="fragment roll-in">Further challenges include presence of multiple copies of exons</p>
            <aside class="notes" data-markdown>
              # RNA

              There is an intermediate step before DNA gets transcribed into mRNA.

              Transcription generates pre-mRNA, the exons within the pre-mRNA are then spliced, removing the introns.

              # RNA-seq

              Sequencing the mRNA.

              Note: What does it produce? Picture of reads?

              Accurate alignment - containing mismatches, insertions and deletions.
            </aside>
          </section>

          <section id="currentAlgo">
            <h2>Background</h2>
            <h3 class="fragment roll-in">Current Algorithmic Solutions</h3>
            <p class="fragment roll-in">Offer compromises in either <span class="fragment highlight-red">accuracy</span> or <span class="fragment highlight-red">resources required</span></p>
            <p class="fragment roll-in">Computational component becoming bottleneck</p>
            <p class="fragment roll-in">Mostly designed for short reads (&le; 200 bases)</p>
            <p class="fragment roll-in">Not great for "Third Generation Sequencing" (potentially full length reads)</p>

            <aside class="notes" data-markdown>
              # TODO:
            </aside>
          </section>

          <section id="STAR">
            <h2>What is STAR?</h2>
              <p class="fragment roll-in">"Spliced Transcripts Alignment to a Reference" (STAR)</p>
              <p class="fragment roll-in">Specifically designed to align non-contiguous sequences to a reference genome</p>
            <aside class="notes" data-markdown>
              It stands for ...

              It's specifically designed...
            </aside>
          </section>

        </section>

        <section id="algoithm">

          <section id="algoithmIntro">
            <h2>Method</h2>
            <h3>Overview</h3>
            <p class="fragment roll-in">The STAR algorithm consists of 2 main steps:
              <ul>
                <li class="fragment roll-in">Seed Search</li>
                <li class="fragment roll-in">Clustering, stitching &amp; scoring</li>
              </ul>
            </p>
            <aside class="notes" data-markdown>
              2 main steps to STAR; seed and cluster/stitch/score.

              I'll go through each stage and explain what's involved.
            </aside>
          </section>

          <section id="algoithmSeed1">
            <h2>Method</h2>
            <h3>Seed Search Algorithm</h3>
            <p class="fragment roll-in">Searching for seeds involves a sequential search for the Maximum Mappable Prefix ($MMP$).</p>
            <p class="fragment roll-in">
              The $MMP$ is calculated as follows:
              <blockquote class="fragment roll-in">
                Given a read sequence $R$, read location $i$ and a reference genome sequence $G$, the $MMP(R,i,G)$ is defined as the longest substring $(R_i,R_{i+1},\dots,R_{i+MML-1})$ that matches exactly one or more substrings of $G$, where $MML$ is the Maximum Mappable Length.
              </blockquote>
            </p>

            <aside class="notes" data-markdown>
              The main idea behind the seed search phase of STAR is to sequentially search through looking for the Maximum Mappable Prefix.

              [read the algo]

              That didn't make a whole lot of sense to me straight away so I'll demostrate this with a diagram...
            </aside>
          </section>

          <section id="algoithmSeed2">
            <h2>Method</h2>
            <h3>Seed Search Example</h3>
            <p class="fragment fade">
              <img width="400px" align="right" src="images/fig1.png">
            </p>
            <p>
              <video width="480px" height="300px" controls>
                <source src="video/mmp.mov" type="video/mp4">
                Your browser does not support the video tag.
              </video>
            </p>
            <br />
            <p class="fragment roll-in">
              The search is implemented using a suffix array for the reference genome; the read sequence is then threaded through.
            </p>

            <aside class="notes" data-markdown>
              ## Basic Idea
              - Once the MMP is found, this is a splice location, rerun this on the remainder of the read sequence that is not mapped [show diagram].

              - This approach can also help finding mismatches/indels. Run a sequence, if it doesn't reach the end, expand the "anchor" allowing for mismatches (what about indels?).

              - Poor alignment from this procedure can help identify sequence traits such as poly-A tails (+AAAAAAA), library adapters (???) or just poor sequencing.

              - Search is performed forwards and backwards on the read, and also can be started from a user defined position to facilitate finding anchors with errors at the ends.

              - The uncompressed SA offer a speed advantage with a trade off for space. Which we'll come to in the results section.

              - Build a suffix array (slightly more efficient that a suffix tree) of the reference genome and thread the read sequence through the array for a given read position.

              TODO: GO OVER suffix arrays?
            </aside>
          </section>

          <section id="algoithmClustering">
            <h2>Method</h2>
            <h3>Clustering, Stitching and Scoring</h3>
            <p class="fragment roll-in">Cluster seeds around a selected anchor seed.</p>
            <p class="fragment roll-in">Anchor seeds are selected by minimising the number of genomic mappings.</p>
            <video class="fragment roll-in" width="480px" height="300px" controls>
              <source src="video/clust.mov" type="video/mp4">
              Your browser does not support the video tag.
            </video>

            <aside class="notes" data-markdown>
              STAR next builds the entire read sequence...

              Clustering the seeds found in the first stage by proximity to a selected number of anchor seeds.

              Anchors are selected from the seeds that have the minimum number of genomic mappings. This prioritises seeds that are not as "multi-mapped".

              The seeds that map within a user-defined genomic window are stiched together using the local-linear transcription model (genomic ordering of local seeds maintained, with no overlaps).

              The size of the genomic window determines the max introm size for spliced allignments.

              If one window is not enough to cover the read sequence, then the alignment is expanded to include another window - resulting in a chimeric read. A join of 2 separate exons from potentially different parts of the genome (diff. chromosome?), joining to form at RNA.

              The stitching is guided by a local alignment scoring scheme that includes user-defined scores for mis/matches and indels.
            </aside>
          </section>
        </section>

        <section id="results">

          <section id="resultsIntro">
            <h2>Results</h2>
            <h3>Method</h3>
            <p class="fragment roll-in">The paper compared STAR 2.1.3 results with 4 other popular RNA-seq mappers:
              <ul>
                <li class="fragment roll-in">TopHat 2.0.0</li>
                <li class="fragment roll-in">GSNAP 2012-07-03</li>
                <li class="fragment roll-in">RUM 1.11</li>
                <li class="fragment roll-in">MapSplice</li>
              </ul>
            </p>

            <aside class="notes" data-markdown>
              All the testing was completed and compared with 4 other RNA-seq mappers.
            </aside>
          </section>

          <section id="resultsSimulated">
            <h2>Results</h2>
            <h3>Performance on Simulated Data</h3>
            <img class="fragment fade" width="420px" align="right" src="images/fig2.png">
            <p class="fragment roll-in">Simulated data allows for accurate expected results.</p>
            <p class="fragment roll-in">
              All aligners were run in <em class="fragment highlight-red">de novo</em> mode with <em class="fragment highlight-red">default parameters</em>.
            </p>
            <p class="fragment roll-in">
              "All aligners exhibit desirable steep ROC curves at high values of detection threshold."
            </p>
            <p class="fragment roll-in">
              "At the lowest detection threshold of 1 read per junction, STAR exhibits the lowest false-positive rate while achieving high sensitivity."
            </p>

            <aside class="notes" data-markdown>
              Initially, they used simulated data to accurately benchmark all the mappers; they know exactly what mapping to expect.

              __de novo__
              - from the beginning - without any knowledge of the genome/transcript. -- String comparison.

              __default parameters__
              - commonly accepted practice, since all aligners should have been optimised for mammalian genomes and RNA-seq data.

            </aside>
          </section>

          <section id="results2">
            <h2>Results</h2>
            <h3>Performance on Experimental Data</h3>
            <img src="images/fig3.png">           
            <p>
              Write about the results...            
            </p>

            <aside class="notes" data-markdown>
              Write some speaker notes...
            </aside>
          </section>

          <section id="resultsSpeed">
            <h2>Results</h2>
            <h3>Speed Comparison</h3>
            <p class="fragment roll-in">Tests completed on a "two 6-core Intel Xeon CPUs X5680@ 3.33GHz and 148GB of RAM".</p>
            <img class="fragment roll-in" width="450px" align="right" src="images/table1.png">
            <p class="fragment roll-in">All mappers were run with default parameters on the ~40 million 2 x 76 Illumina human RNA-seq dataset.
            </p>

            <aside class="notes" data-markdown>
              Write some speaker notes...
            </aside>
          </section>

          <section id="resultsExperimental">
            <h2>Results</h2>
            <h3>Experimental Evaluation</h3>
            <img src="images/table2.png">           
            <p>
              Write about the results...            
            </p>

            <aside class="notes" data-markdown>
              Write some speaker notes...
            </aside>
          </section>

        </section>

        <section id="discussion">
          <h2>Discussion</h2>
          <p>Aligning non-contiguous RNA-seq data to a reference genome is hard. It's still unsolved.</p>
          <p>STAR is a stand-alone C++ RNA-seq mapper.</p>

          <aside class="notes" data-markdown>
            - Stand-alone RNA-seq mapper -- E.G. not a modified DNA mapper.
            - 

          </aside>
        </section>

      </div>
    </div>

    <script src="lib/js/head.min.js"></script>
    <script src="js/reveal.min.js"></script>

    <script>

      // Full list of configuration options available here:
      // https://github.com/hakimel/reveal.js#configuration
      Reveal.initialize({
        controls: true,
        progress: true,
        history: true,
        center: false,
        overview: false,
        theme: Reveal.getQueryHash().theme, // available themes are in /css/theme
        transition: Reveal.getQueryHash().transition || 'none', // default/cube/page/concave/zoom/linear/fade/none
        slideNumber: true,

        // Parallax scrolling
        // parallaxBackgroundImage: 'images/background-grass.jpg',
        // parallaxBackgroundSize: '2100px 900px',
        // parallaxBackgroundImage: 'images/background-snow.jpg',
        // parallaxBackgroundSize: '2560px 1440px',
        // parallaxBackgroundImage: 'images/background-lights.jpg',
        // parallaxBackgroundSize: '2560px 1600px',
        // parallaxBackgroundImage: 'images/background-stars.jpg',
        // parallaxBackgroundSize: '4000px 2667px',
        // parallaxBackgroundImage: 'images/background-stars2.jpg',
        // parallaxBackgroundSize: '1920px 1280px',
        // parallaxBackgroundImage: 'images/background-stars3.jpg',
        // parallaxBackgroundSize: '2560px 1600px',

        // Optional libraries used to extend on reveal.js
        dependencies: [
          { src: 'lib/js/classList.js', condition: function() { return !document.body.classList; } },
          { src: 'plugin/markdown/marked.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
          { src: 'plugin/markdown/markdown.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
          { src: 'plugin/highlight/highlight.js', async: true, callback: function() { hljs.initHighlightingOnLoad(); } },
          { src: 'plugin/zoom-js/zoom.js', async: true, condition: function() { return !!document.body.classList; } },
          { src: 'plugin/notes/notes.js', async: true, condition: function() { return !!document.body.classList; } },
          { src: 'plugin/math/math.js', async: true }
        ]
      });

    </script>

  </body>
</html>
